language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -290,17 +301,36 @@ public BallerinaTextDocumentService(BallerinaLanguageServer ballerinaLanguageSer
     @Override
     public CompletableFuture<List<? extends TextEdit>> formatting(DocumentFormattingParams params) {
         return CompletableFuture.supplyAsync(() -> {
+            ScriptEngine engine = new ScriptEngineManager().getEngineByName(JAVASCRIPT_ENGINE);
             TextDocumentServiceContext formatContext = new TextDocumentServiceContext();
             formatContext.put(DocumentServiceKeys.FILE_URI_KEY, params.getTextDocument().getUri());
             String fileContent = documentManager.getFileContent(CommonUtil.getPath(params.getTextDocument().getUri()));
+            String newDummyContent = fileContent;
             String[] contentComponents = fileContent.split("\\n|\\r\\n|\\r");
             int lastNewLineCharIndex = Math.max(fileContent.lastIndexOf("\n"), fileContent.lastIndexOf("\r"));
             int lastCharCol = fileContent.substring(lastNewLineCharIndex + 1).length();
             int totalLines = contentComponents.length;
             Range range = new Range(new Position(0, 0), new Position(totalLines, lastCharCol));
             JsonObject ast = TextDocumentFormatUtil.getAST(params, documentManager, formatContext);
-            // TODO: For the moment we replace the content with the ast. Need to implement the AST source-gen
-            String newDummyContent = ast.toString();
+            ClassLoader classLoader = BallerinaTextDocumentService.class.getClassLoader();
+            InputStream inputStream = classLoader.getResourceAsStream(SOURCE_GEN_FILE);
+            try {
+                Reader fileReader = new InputStreamReader(inputStream, "UTF-8");
+                engine.eval(fileReader);
+                fileReader.close();
+                Invocable invocable = (Invocable) engine;
+                newDummyContent = (String) invocable.invokeFunction("gen", ast.toString());
```

Comment:
Should this content still be called dummy ? It’s proper content now right?

Reply:
No dummy it should not 😀 will change it.